modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/snapshot/storage/SnapshotLocalFileSystemStorage.scala

```diff
@@ -97,11 +103,50 @@ abstract class SnapshotLocalFileSystemStorage[
       link(hashName, toOrdinalName(snapshot))
     }
 
+  def findAbove(ordinal: SnapshotOrdinal): F[Stream[F, File]] = {
+    val baseDirectory = (ordinal.value.value / ordinalChunkSize.value) * ordinalChunkSize.value
+
+    def isAbove(file: File): Boolean = {
+      val fileOrdinal = file.name.toLong
+      fileOrdinal > ordinal.value.value
+    }
+
+    def listFilesFrom(base: Long): F[Stream[F, File]] =
+      dir
+        .map(_ / base.toString)
+        .flatMap { a =>
+          Async[F].blocking {
+            if (base === baseDirectory) a.list(isAbove(_)) else a.list
+          }
+        }
+        .map(Stream.fromIterator[F](_, 1))
+
+    def go(base: Long): F[Stream[F, File]] =
+      dir
+        .map(a => a / base.toString)
+        .map(_.isEmpty)
+        .ifM(
+          Async[F].pure(Stream.empty),
+          listFilesFrom(base).flatMap { filesInBase =>
+            val nextBase = base + ordinalChunkSize.value
+            go(nextBase).map { filesInNextBase =>
+              filesInBase ++ filesInNextBase
+            }
+          }
+        )
+
+    go(baseDirectory)
+  }
+
   private def toOrdinalName(snapshot: S): String = toOrdinalName(snapshot.ordinal)
-  private def toOrdinalName(ordinal: SnapshotOrdinal): String = ordinal.value.value.toString
+  private def toOrdinalName(ordinal: SnapshotOrdinal): String =
+    "ordinal/" + ordinalPathGenerator.get(ordinal.value.value.toString)
```

Comment:
Instead 
ordinal.value.value.toString
maybe
ordinal.show?

Reply:
No, `show` is for showing/logging purpose. It adds additional formatting to the string that we do not want here